File: 3866-first-unique-even-element/3866-first-unique-even-element.cs
public class Solution {
    public int FirstUniqueEven(int[] nums) {
        int n = nums.Length;
        var map = new int[101];

        foreach(var num in nums)
            map[num]++;
        for (int i = 0; i < n; ++i) {
            if ((nums[i] & 1) == 0 && map[nums[i]] == 1)
                return nums[i];
        }
        return -1;
    }
}

File: 3866-first-unique-even-element/3866-first-unique-even-element.cpp
using vi = vector<int>;

class Solution {
public:
    int firstUniqueEven(vector<int>& nums) {
        int n = size(nums);
        vi map(101, 0);
        for (int i = 0; i < n; ++i)
            map[nums[i]]++;

        for (int i = 0; i < n; ++i) {
            if (nums[i] & 1)
                continue;
            
            if (map[nums[i]] == 1)
                return nums[i];
        }
        return -1;
    }
};